server/room.ts

```diff
@@ -29,212 +36,124 @@ export class Room {
       this.deserialize(roomData);
     }
 
-    setInterval(() => {
-      // console.log(roomId, this.video, this.roster, this.tsMap, this.nameMap);
+    this.roomInterval = setInterval(() => {
       io.of(roomId).emit('REC:tsMap', this.tsMap);
+      if (
+        this.videoDuration !== 0 &&
+        this.videoDuration &&
+        this.videoTS >= this.videoDuration
+      ) {
+        setTimeout(() => {
+          this.nextVideo();
+        }, 1000);
+      }
     }, 1000);
 
     io.of(roomId).on('connection', (socket: Socket) => {
-      // console.log(socket.id);
-      this.roster.push({ id: socket.id });
+      this.connections.push(new Connection(socket, this));
+    });
+  }
 
-      socket.emit('REC:host', this.getHostState());
-      socket.emit('REC:nameMap', this.nameMap);
-      socket.emit('REC:pictureMap', this.pictureMap);
-      socket.emit('REC:tsMap', this.tsMap);
-      socket.emit('chatinit', this.chat);
-      io.of(roomId).emit('roster', this.roster);
+  removeConnection = (socketId: string) => {
+    this.connections = this.connections.filter(
```

Comment:
I believe this updates the reference to this.connections, so splicing out the removal is probably better.

Reply:
done

Reply:
Isn't this still reassigning?